openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java

```diff
@@ -414,6 +433,79 @@ public Response search(SearchRequest request) throws IOException {
     }
   }
 
+  public List<?> buildSearchHierarchy(SearchRequest request, SearchResponse searchResponse) {
+    List<?> response = new ArrayList<>();
+    if (request.getIndex().equalsIgnoreCase("glossary_term_search_index")) {
+      response = buildGlossaryTermSearchHierarchy(searchResponse);
+    }
+    return response;
+  }
+
+  public List<GlossaryTerm> buildGlossaryTermSearchHierarchy(SearchResponse searchResponse) {
+    Map<String, GlossaryTerm> termMap = new LinkedHashMap<>(); // termMap represent glossary terms
```

Comment:
any reason we need LinkedHashMap here?

Reply:
yes.. we need to maintain the order of terms by fqn to get correct hierarchical  results